ui/src/components/userProfile/UserProfileDesktop.js

```diff
@@ -116,12 +119,29 @@ const UserProfileDesktop = (props) => {
   }, [user.id]);
 
   const fetchPublishedForms = useCallback(async () => {
+    //get all forms
     const { data, status } = await getLoginOnlyFormsRequest();
-
     if (status === 200) {
-      setFormList([...data]);
+      setFormList([...data])
+
+    //find forms the user have signed up for
+      const tempFormList = data
```

Comment:
i think this needs to be a backend API call. UI shouldn't be doing any of these logics.